### PR TITLE
[Merged by Bors] - Implement Annex-B string html methods

### PR DESCRIPTION
**Pull request by HalidOdat**
Implements all Annex-B string html methods, they are deprecated but still part of the ES5 spec. They have been feature gated under the `"annex-b"` feature flag

### CODE REVIEW

File: boa_engine/src/builtins/string/mod.rs [hunk 1]
```diff
@@ -2165,6 +2186,322 @@ impl String {
...
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-createhtml
+    pub(crate) fn create_html(
+        string: &JsValue,
+        tag: &[u16],
+        attribute_and_value: Option<(&[u16], &JsValue)>,
+        context: &mut Context<'_>,
+    ) -> JsResult<JsValue> {
+        // 1. Let str be ? RequireObjectCoercible(string).
+
```
Comment (raskad) [inline boa_engine/src/builtins/string/mod.rs:2204]:
```suggestion
```

File: boa_engine/src/builtins/string/mod.rs [hunk 2]
```diff
@@ -2165,6 +2186,322 @@ impl String {
...
+        let str = string.require_object_coercible()?;
+
+        // 2. Let S be ? ToString(str).
+        let s = str.to_string(context)?;
+
+        // 3. Let p1 be the string-concatenation of "<" and tag.
+        let mut p1 = JsString::concat_array(&[utf16!("<"), tag]);
+        // 4. If attribute is not the empty String, then
```
Comment (raskad) [inline boa_engine/src/builtins/string/mod.rs:2212]:
```suggestion
        let mut p1 = JsString::concat_array(&[utf16!("<"), tag]);

        // 4. If attribute is not the empty String, then
```

File: boa_engine/src/builtins/string/mod.rs [hunk 3]
```diff
@@ -2165,6 +2186,322 @@ impl String {
...
+            utf16!(">"),
+            s.as_slice(),
+            utf16!("</"),
+            tag,
+            utf16!(">"),
+        ]);
+        // 8. Return p4.
+        Ok(p4.into())
+    }
+    /// `String.prototype.anchor( name )`
```
Comment (raskad) [inline boa_engine/src/builtins/string/mod.rs:2261]: Let's add an empty line between all of the functions to make the code less cluttered.

```suggestion
    }

    /// `String.prototype.anchor( name )`
```

**Review by raskad — APPROVED** [review]
Just got some style suggestions. Nice addition!

File: boa_engine/src/builtins/string/mod.rs [hunk 4]
```diff
@@ -2165,6 +2186,322 @@ impl String {
     }
 }
 
+#[cfg(feature = "annex-b")]
+impl String {
+    /// Helper function for creating html elements.
```
Comment (Razican) [inline boa_engine/src/builtins/string/mod.rs:2191]: Here I would add the definition in the spec, as we do in other functions:

```
B.2.2.2.1 CreateHTML(string, tag, attribute, value)

The abstract operation...
```

**Review by Razican — COMMENTED** [review]
Looks great! I added a suggestion :)

**Review by Razican — APPROVED** [review]

**Pull request closed without merging**